scripts/SREP/retry-pruning-cronjob/script.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+# Delete jobs that fail in openshift-sre-pruning namespace.
+
+set -x
+set -e
+set -o nounset
+set -o pipefail
+
+readonly PRUNING_NS=openshift-sre-pruning
+FailedJobs=()
+
+detect_job() {
+  echo "INFO: finding jobs producing error"
+  for current_ns in ${PRUNING_NS}; do
+    readarray -t Jobs < <(oc get job -n "$current_ns" | awk '{print $1}') 
+    Jobs=("${Jobs[@]:1}")
+    for job in "${Jobs[@]}"; do
+      JOB_STATUS=$(oc get job "$job" -n "$current_ns" -o jsonpath={.status.succeeded})
+      if [[ $JOB_STATUS -ne 1 ]]; then
+        echo "Found $job is failing"
+        FailedJobs+=("$job")
+      fi
+    done
+  done
+}
+
+delete_job() {
+  if [[ ${#FailedJobs[@]} == 0 ]]; then
+    echo "INFO: no failed jobs found"
+    return 1
+  else
+    echo "INFO: deleting jobs producing error"
+    for jobs in "${FailedJobs[@]}"; do
+      oc delete job "$jobs" -n "$PRUNING_NS"
+    done
+  fi
+}
+
+main() {
```

Comment:
We should retry the same failed job after deleted here.

Reply:
yes, once would be enough, if retry fails, that means there are two failures in total, it should not be an intermittent issue in that case.